File: server-side/Core/Models/Comment.cs
using System.Collections.Generic;

namespace Core.Models
{
    public class Comment : BaseEntity
    {
        public int UserId { get; set; }
        public int BlogId { get; set; }
        public string Text { get; set; }
        public bool IsReply { get; set; }

        public User User { get; set; }
        public Blog Blog { get; set; }
        public ICollection<CommentReply> CommentReplies { get; set; }
    }
}

File: client-side/src/app/shared/models/comment.ts
import { User } from './user';

export interface Comment {
  id: number;
  blogId: number;
  text: string;
  isReply: boolean;
  userDTO: User;
  addedDate: Date;
  commentReplyDTOs: CommentReply[];
}

export class CommentFormValues {
  text: string;
  slug: string;
  email: string;

  constructor(text: string, slug: string, email: string) {
    this.text = text;
    this.slug = slug;
    this.email = email;
  }
}

export interface CommentReply {
  text: string;
  commentId: number;
  addedDate: Date;
  userDTO: User;
}

export class CommentReplyFormValues {
  text: string;
  slug: string;
  email: string;
  commentId: number;

  constructor(text: string, slug: string, email: string, commentId: number) {
    this.text = text;
    this.slug = slug;
    this.email = email;
    this.commentId = commentId;
  }
}
